compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -883,6 +883,34 @@ object Checking {
     templ.parents.find(_.tpe.derivesFrom(defn.PolyFunctionClass)) match
       case Some(parent) => report.error(s"`PolyFunction` marker trait is reserved for compiler generated refinements", parent.srcPos)
       case None =>
+
+  /** check that parameters of a java defined annotations are all named arguments if we have more than one parameter */
+  def checkNamedArgumentForJavaAnnotation(annot: untpd.Tree, sym: ClassSymbol)(using Context): untpd.Tree = 
+    assert(sym.is(JavaDefined))
+
+    def annotationHasValueField: Boolean = 
+      sym.info.decls.exists(_.name == nme.value)
+
+    def hasOnePositionalParameter(params: List[untpd.Tree]): Boolean = 
+      params.exists(!_.isInstanceOf[untpd.NamedArg])
+
+    def mapPositionalToNamed(param: untpd.Tree): untpd.Tree = param match
+      case tree: untpd.NamedArg => tree
+      case tree => untpd.cpy.NamedArg(param)(nme.value, param)
+    end mapPositionalToNamed
+
+    annot match
+      case untpd.Apply(fun, params) if hasOnePositionalParameter(params) && annotationHasValueField => 
+        untpd.cpy.Apply(annot)(fun, params.mapConserve(mapPositionalToNamed))
```

Comment:
This does not seem to match the spec anymore. It would allow `@JavaAnnot(1, other = 2)`, if there is `value` and `other`, which should not be allowed. Also it's probably going to do the wrong thing for `@JavaAnnot(1, 2)` if there is a `value` field, as it will transform them both to `@JavaAnnot(value = 1, value = 2)`.

The auto-`value =` treatment should only be considered with `params.size == 1`.

Reply:
Mmmh, I was pretty sure it worked with Scala 2.13.13 when I tried yesterday 😅 . I will revert it.